components/esp_lcd/src/esp_lcd_panel_rgb.c

```diff
@@ -172,9 +172,11 @@ static esp_err_t lcd_rgb_panel_alloc_frame_buffers(const esp_lcd_rgb_panel_confi
     return ESP_OK;
 }
 
-static esp_err_t lcd_rgb_panel_destory(esp_rgb_panel_t *rgb_panel)
+static esp_err_t lcd_rgb_panel_destroy(esp_rgb_panel_t *rgb_panel)
 {
-    lcd_ll_enable_clock(rgb_panel->hal.dev, false);
+    if (rgb_panel->hal.dev) {
+        lcd_ll_enable_clock(rgb_panel->hal.dev, false);
+    }
```

Comment:
sure, makes sense.  It looks like a crash could occur here if the framebuffer allocation failed. I'm guessing that's what you're fixing?

Reply:
Correct! The clock is enabled and `hal` set after the framebuffer allocation is attempted.